common/src/state/data_transfer.rs

```diff
@@ -0,0 +1,258 @@
+                name: _,
+                current,
+                total,
+            } => TransferProgress::Progress(
+                total
+                    .map(|total| current as f64 / total as f64 * 100.)
+                    .unwrap_or_default() as u8,
+            ),
+            Progression::ProgressComplete { name: _, total: _ } => TransferProgress::Finishing,
+            Progression::ProgressFailed {
+                name: _,
+                last_size: _,
+                error: _,
+            } => TransferProgress::Error,
+        };
+        if let Some(f) = self
+            .get_tracker_from(tracker)
+            .iter_mut()
+            .find(|p| file_id.eq(&p.id))
+        {
+            f.progress = progress;
+        }
+    }
+
+    pub fn update_file_description(
+        &mut self,
+        file_id: &Uuid,
+        description: String,
+        tracker: TrackerType,
+    ) {
+        if let Some(f) = self
+            .get_tracker_from(tracker)
+            .iter_mut()
+            .find(|p| file_id.eq(&p.id))
+        {
+            f.description = Some(description);
+        }
+    }
+
+    pub fn pause_file_upload(&mut self, file_id: &Uuid, tracker: TrackerType) {
+        if let Some(f) = self
+            .get_tracker_from(tracker)
+            .iter_mut()
+            .find(|p| file_id.eq(&p.id))
+        {
+            let current = if let TransferProgress::Progress(p) = f.progress {
+                p
+            } else {
+                0
+            };
+            f.progress = TransferProgress::Paused(current);
+        }
+    }
+
+    pub fn cancel_file_upload(&mut self, file_id: &Uuid, tracker: TrackerType) {
+        if let Some(f) = self
+            .get_tracker_from(tracker)
+            .iter_mut()
+            .find(|p| file_id.eq(&p.id))
+        {
+            f.progress = TransferProgress::Cancelling;
+        }
+    }
+
+    pub fn error_file_upload(&mut self, file_id: &Uuid, tracker: TrackerType) {
+        if let Some(f) = self
+            .get_tracker_from(tracker)
+            .iter_mut()
+            .find(|p| file_id.eq(&p.id))
+        {
+            f.progress = TransferProgress::Error;
+            f.description = Some(get_local_text("files.error-to-upload"));
+        }
+    }
+
+    pub fn remove_file_upload(&mut self, file_id: &Uuid, tracker: TrackerType) {
+        self.get_tracker_from(tracker)
+            .retain(|p| !file_id.eq(&p.id))
+    }
+
+    fn get_tracker_from(&mut self, tracker: TrackerType) -> &mut Vec<FileProgress> {
+        match tracker {
+            TrackerType::FileUpload => &mut self.file_progress_upload,
+            TrackerType::FileDownload => &mut self.file_progress_download,
+        }
+    }
+
+    pub fn get_tracker(&self, upload: bool) -> Vec<FileProgress> {
+        if upload {
+            self.file_progress_upload.clone()
+        } else {
+            self.file_progress_download.clone()
+        }
+    }
```

Comment:
We should borrow from it instead of cloning.

```suggestion
    pub fn get_tracker(&self, upload: bool) -> &[FileProgress] {
        if upload {
            &self.file_progress_upload
        } else {
            &self.file_progress_download
        }
    }
```